data_safe_haven/infrastructure/programs/sre/gitea_external_server.py

```diff
@@ -0,0 +1,349 @@
+from collections.abc import Mapping
+
+from pulumi import ComponentResource, Input, Output, ResourceOptions
+from pulumi_azure_native import containerinstance, storage
+
+from data_safe_haven.infrastructure.common import (
+    DockerHubCredentials,
+    get_ip_address_from_container_group,
+)
+from data_safe_haven.infrastructure.components import (
+    FileShareFile,
+    FileShareFileProps,
+    LocalDnsRecordComponent,
+    LocalDnsRecordProps,
+    PostgresqlDatabaseComponent,
+    PostgresqlDatabaseProps,
+)
+from data_safe_haven.resources import resources_path
+from data_safe_haven.utility import FileReader
+
+
+class SREGiteaExternalServerProps:
+    """Properties for SREGiteaExternalServerComponent"""
+
+    def __init__(
+        self,
+        containers_subnet_id: Input[str],
+        database_password: Input[str],
+        database_subnet_id: Input[str],
+        dns_server_ip: Input[str],
+        dockerhub_credentials: DockerHubCredentials,
+        ldap_server_hostname: Input[str],
+        ldap_server_port: Input[int],
+        ldap_username_attribute: Input[str],
+        ldap_user_filter: Input[str],
+        ldap_user_search_base: Input[str],
+        location: Input[str],
+        resource_group_name: Input[str],
+        sre_fqdn: Input[str],
+        storage_account_key: Input[str],
+        storage_account_name: Input[str],
+        database_username: Input[str] | None = None,
+    ) -> None:
+        self.containers_subnet_id = containers_subnet_id
+        self.database_password = database_password
+        self.database_subnet_id = database_subnet_id
+        self.database_username = (
+            database_username if database_username else "postgresadmin"
+        )
+        self.dns_server_ip = dns_server_ip
+        self.dockerhub_credentials = dockerhub_credentials
+        self.ldap_server_hostname = ldap_server_hostname
+        self.ldap_server_port = ldap_server_port
+        self.ldap_username_attribute = ldap_username_attribute
+        self.ldap_user_filter = ldap_user_filter
+        self.ldap_user_search_base = ldap_user_search_base
+        self.location = location
+        self.resource_group_name = resource_group_name
+        self.sre_fqdn = sre_fqdn
+        self.storage_account_key = storage_account_key
+        self.storage_account_name = storage_account_name
+
+
+class SREGiteaExternalServerComponent(ComponentResource):
+    """Deploy Gitea server with Pulumi"""
```

Comment:
There's a lot of overlap between this and the `SREGiteaServerComponent`. Can we combine both into a single reusable component?

Reply:
Yes, that's what I'm actually doing, I just haven't pushed yet